### PR TITLE
add support for kyria keyboard

### PR DESCRIPTION
**Pull request by nosarthur**

### CODE REVIEW

File: keyboards/splitkb/kyria/keymaps/vial/rules.mk [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no          # Audio output
+SPLIT_KEYBOARD = yes       # Split common
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
+
+DEFAULT_FOLDER = splitkb/kyria/rev2
+
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/keymaps/vial/rules.mk]:
```suggestion
ENCODER_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```
Most this stuff is replicated from the keyboard level.
The Vial keymap in the main repo shouldn't overwrite the bootloader.

**Review by lesshonor — CHANGES_REQUESTED** [review]
You need to move this keymap under `splitkb/kyria/rev2`, because rev1 and rev2 don't have the same matrix:
https://github.com/vial-kb/vial-qmk/blob/8356266e078cefeec7c1e4a8c1d59118ad5a0755/keyboards/splitkb/kyria/rev1/rev1.h#L41
https://github.com/vial-kb/vial-qmk/blob/8356266e078cefeec7c1e4a8c1d59118ad5a0755/keyboards/splitkb/kyria/rev2/rev2.h#L41

File: keyboards/splitkb/kyria/keymaps/vial/rules.mk (hunk 1 above)
Comment (nosarthur) [reply keyboards/splitkb/kyria/keymaps/vial/rules.mk]: I have deleted some options that are identical to the keyboard level. Some are left here to reduce the firmware size

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# MCU name
+MCU = atmega32u4
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]:
```suggestion
```
https://github.com/vial-kb/vial-qmk/blob/4eb1c2e21fef3b2a94215992158c0195b92a8d4d/keyboards/splitkb/kyria/rules.mk#L2
This is already set to the exact same value at the keyboard level.

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+# Bootloader selection
+BOOTLOADER = caterina
+
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]:
```suggestion
```
This is set at the keyboard level: https://github.com/vial-kb/vial-qmk/blob/4eb1c2e21fef3b2a94215992158c0195b92a8d4d/keyboards/splitkb/kyria/rules.mk#L5
Undoing this to put `caterina` here means `QK_BOOT` and flashing from the command line won't work the same way it does for all other builds of the Kyria. Your personal keymap, sure, but this is not an appropriate change for the de-facto Vial keymap.

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes 		# Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]: If you're really concerned about file size, leaving mousekeys disabled frees up around 1800 bytes.

```suggestion
MOUSEKEY_ENABLE = yes 		# Mouse keys
```
`BOOTMAGIC_ENABLE` is forcibly set to yes if `VIA_ENABLE` is set to `yes`, so attempting to set it to `no` here does nothing: https://github.com/vial-kb/vial-qmk/blob/4eb1c2e21fef3b2a94215992158c0195b92a8d4d/builddefs/common_features.mk#L609-L615

`EXTRAKEY_ENABLE = yes` is already set at the keyboard level: https://github.com/vial-kb/vial-qmk/blob/4eb1c2e21fef3b2a94215992158c0195b92a8d4d/keyboards/splitkb/kyria/rules.mk#L12

`SLEEP_LED_ENABLE` is disabled by default, so attempting to set it to `no` here does nothing: https://github.com/vial-kb/vial-qmk/blob/4eb1c2e21fef3b2a94215992158c0195b92a8d4d/builddefs/common_features.mk#L530-L535

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+
+
+ENCODER_ENABLE = yes
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]:
```suggestion
```
This is already set at the revision level: https://github.com/vial-kb/vial-qmk/blob/4eb1c2e21fef3b2a94215992158c0195b92a8d4d/keyboards/splitkb/kyria/rev2/rules.mk#L3

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+AUTO_SHIFT_ENABLE = yes
+COMBO_ENABLE = yes
+TAP_DANCE_ENABLE = no
+
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
+
+QMK_SETTINGS = yes
+KEY_OVERRIDE_ENABLE = no
+DEFERRED_EXEC_ENABLE = no
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]:
```suggestion
```
This is disabled by default.

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+SPACE_CADET_ENABLE = no
+GRAVE_ESC_ENABLE = no
+MAGIC_ENABLE = no
+AVR_USE_MINIMAL_PRINTF = yes
+DYNAMIC_MACRO_ENABLE = no
+MUSIC_ENABLE = no
+SWAP_HANDS_ENABLE= no
+
```
Comment (lesshonor) [inline keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]:
```suggestion
```
These are also disabled by default.

**Review by lesshonor — CHANGES_REQUESTED** [review]

File: keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk (hunk 4 above)
Comment (nosarthur) [reply keyboards/splitkb/kyria/rev2/keymaps/vial/rules.mk]: good to know, thanks. I was desperate to try various options to reduce the firmware size

**Review by lesshonor — APPROVED** [review]

**Pull request merged**